unittests/misc_tests.cpp

```diff
@@ -122,21 +90,43 @@ BOOST_AUTO_TEST_CASE(reverse_endian_tests)
-   BOOST_CHECK_EQUAL( name{name_suffix("abc..def"_n)}, name{"def"_n} );
-   BOOST_CHECK_EQUAL( name{name_suffix("abc.def.ghi"_n)}, name{"ghi"_n} );
-   BOOST_CHECK_EQUAL( name{name_suffix(".abcdefghij"_n)}, name{"abcdefghij"_n} );
-   BOOST_CHECK_EQUAL( name{name_suffix(".abcdefghij.1"_n)}, name{"1"_n} );
-   BOOST_CHECK_EQUAL( name{name_suffix("a.bcdefghij"_n)}, name{"bcdefghij"_n} );
-   BOOST_CHECK_EQUAL( name{name_suffix("a.bcdefghij.1"_n)}, name{"1"_n} );
-   BOOST_CHECK_EQUAL( name{name_suffix("......a.b.c"_n)}, name{"c"_n} );
-   BOOST_CHECK_EQUAL( name{name_suffix("abcdefhi.123"_n)}, name{"123"_n} );
-   BOOST_CHECK_EQUAL( name{name_suffix("abcdefhij.123"_n)}, name{"123"_n} );
+   BOOST_CHECK_EQUAL( name(0).suffix(), name{0} );
+   BOOST_CHECK_EQUAL( name{"eosio"_n}.suffix(), name{"eosio"_n} );
+   BOOST_CHECK_EQUAL( name{"eosio.any"_n}.suffix(), name{"any"_n} );
+   BOOST_CHECK_EQUAL( name{"abcdehijklmn"_n}.suffix(), name{"abcdehijklmn"_n} );
+   BOOST_CHECK_EQUAL( name{"abcdehijklmn1"_n}.suffix(), name{"abcdehijklmn1"_n} );
+   BOOST_CHECK_EQUAL( name{"abc.def"_n}.suffix(), name{"def"_n} );
+   BOOST_CHECK_EQUAL( name{".abc.def"_n}.suffix(), name{"def"_n} );
+   BOOST_CHECK_EQUAL( name{"..abc.def"_n}.suffix(), name{"def"_n} );
+   BOOST_CHECK_EQUAL( name{"abc..def"_n}.suffix(), name{"def"_n} );
+   BOOST_CHECK_EQUAL( name{"abc.def.ghi"_n}.suffix(), name{"ghi"_n} );
+   BOOST_CHECK_EQUAL( name{".abcdefghij"_n}.suffix(), name{"abcdefghij"_n} );
+   BOOST_CHECK_EQUAL( name{".abcdefghij.1"_n}.suffix(), name{"1"_n} );
+   BOOST_CHECK_EQUAL( name{"a.bcdefghij"_n}.suffix(), name{"bcdefghij"_n} );
+   BOOST_CHECK_EQUAL( name{"a.bcdefghij.1"_n}.suffix(), name{"1"_n} );
+   BOOST_CHECK_EQUAL( name{"......a.b.c"_n}.suffix(), name{"c"_n} );
+   BOOST_CHECK_EQUAL( name{"abcdefhi.123"_n}.suffix(), name{"123"_n} );
+   BOOST_CHECK_EQUAL( name{"abcdefhij.123"_n}.suffix(), name{"123"_n} );
+}
+
+BOOST_AUTO_TEST_CASE(name_suffix_additional_tests) {
+   // ----------------------------
+   // constexpr name suffix()const
+   BOOST_CHECK_EQUAL( name{".eosioaccounj"}.suffix(), name{"eosioaccounj"} );
+   BOOST_CHECK_EQUAL( name{"e.osioaccounj"}.suffix(), name{"osioaccounj"} );
+   BOOST_CHECK_EQUAL( name{"eo.sioaccounj"}.suffix(), name{"sioaccounj"} );
+   BOOST_CHECK_EQUAL( name{"eos.ioaccounj"}.suffix(), name{"ioaccounj"} );
+   BOOST_CHECK_EQUAL( name{"eosi.oaccounj"}.suffix(), name{"oaccounj"} );
+   BOOST_CHECK_EQUAL( name{"eosio.accounj"}.suffix(), name{"accounj"} );
+   BOOST_CHECK_EQUAL( name{"eosioa.ccounj"}.suffix(), name{"ccounj"} );
+   BOOST_CHECK_EQUAL( name{"eosioac.counj"}.suffix(), name{"counj"} );
+   BOOST_CHECK_EQUAL( name{"eosioacc.ounj"}.suffix(), name{"ounj"} );
+   BOOST_CHECK_EQUAL( name{"eosioacco.unj"}.suffix(), name{"unj"} );
+   BOOST_CHECK_EQUAL( name{"eosioaccou.nj"}.suffix(), name{"nj"} );
+   BOOST_CHECK_EQUAL( name{"eosioaccoun.j"}.suffix(), name{"j"} );
```

Comment:
Can we have:
` BOOST_CHECK_EQUAL( name{"eosioaccounj."}.suffix(), name{""} );`

Reply:
That is not a valid `name`. Added `BOOST_CHECK_EQUAL( name{"eosioaccounja"}.suffix(), name{""} );`